e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -132,170 +133,174 @@ internal static class TestHelper {
                 context.SshConfig.Host,
                 context.SshConfig.Username);
             client.Connect();
-            context.OutputHelper?.WriteLine("ssh connection successful established");
+            context.OutputHelper?.WriteLine("scp connection successful established");
 
             return client;
         }
 
         /// <summary>
-        /// Delete published_nodes.json file into the OPC Publisher edge module
+        /// Gets the private SSH key from the configuration to connect to the Edge VM
         /// </summary>
-        /// <param name="destinationFilePath">Destination file path</param>
         /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
-        public static void DeletePublishedNodesFile(string destinationFilePath, IIoTPlatformTestContext context) {
+        /// <returns></returns>
+        private static PrivateKeyFile GetPrivateSshKey(IIoTPlatformTestContext context)
+        {
+            context.OutputHelper?.WriteLine("Load private key from environment variable");
+
+            var buffer = Encoding.Default.GetBytes(context.SshConfig.PrivateKey);
+            var privateKeyStream = new MemoryStream(buffer);
+
+            var privateKeyFile = new PrivateKeyFile(privateKeyStream);
+            return privateKeyFile;
+        }
+
+        /// <summary>
+        /// Delete a file on the Edge VM
+        /// </summary>
+        /// <param name="fileName">Filename of the file to delete</param>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        public static Task DeleteFileOnEdgeVM(string fileName, IIoTPlatformTestContext context) {
             var isSuccessful = false;
             using var client = CreateSshClientAndConnect(context);
 
-            var terminal = client.RunCommand("rm " + destinationFilePath);
-            if (string.IsNullOrEmpty(terminal.Error)) {
+            var terminal = client.RunCommand("rm " + fileName);
+
+            if (string.IsNullOrEmpty(terminal.Error) || terminal.Error.ToLowerInvariant().Contains("no such file")) {
                 isSuccessful = true;
             }
             Assert.True(isSuccessful, "Delete file was not successful");
+
+            return Task.CompletedTask;
         }
 
         /// <summary>
-        /// CreateFolder a folder on IoTEdge to store published_nodes.json file
+        /// Create a folder on Edge VM (if not exists)
         /// </summary>
-        /// <param name="folderPath">Destination file path</param>
+        /// <param name="folderPath">Name of the folder to create.</param>
         /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
-        private static void CreateFolderOnIoTEdge(string folderPath, IIoTPlatformTestContext context) {
+        private static Task CreateFolderOnEdgeVM(string folderPath, IIoTPlatformTestContext context) {
```

Comment:
Why do we need the method to return `Task` here if there are no asynchronous calls in it ?